### PR TITLE
swf: Various cleanups

### PR DESCRIPTION
**Pull request by relrelb**

### CODE REVIEW

File: core/src/display_object/movie_clip.rs [hunk 1]
```diff
@@ -2855,7 +2855,7 @@ impl<'gc, 'a> MovieClipData<'gc> {
         context: &mut UpdateContext<'_, 'gc, '_>,
         reader: &mut SwfStream<'a>,
     ) -> DecodeResult {
-        let font = reader.read_define_font_2(2)?;
+        let font = reader.read_define_font_2_or_3(2)?;
```
Comment (Herschel) [inline core/src/display_object/movie_clip.rs:2858]: In retrospect, I'd prefer to name this simply `read_define_font_3` to be less wordy and to indicate the max version it handles. Ditto for any other methods in SWF that do `m_or_n` -- name them `n` for simplicity.

File: swf/src/types.rs [hunk 2]
```diff
@@ -608,6 +608,28 @@ pub struct PlaceObject<'a> {
...
+        const HAS_NAME = 1 << 5;
+        const HAS_CLIP_DEPTH = 1 << 6;
+        const HAS_CLIP_ACTIONS = 1 << 7;
+
+        // PlaceObject3
+        const HAS_FILTER_LIST = 1 << 8;
+        const HAS_BLEND_MODE = 1 << 9;
+        const HAS_CACHE_AS_BITMAP = 1 << 10;
+        const HAS_CLASS_NAME = 1 << 11;
+        const IS_IMAGE = 1 << 12;
```
Comment (Herschel) [inline swf/src/types.rs:627]: maybe we should rename this `HAS_IMAGE` since that is the naming used in the spec?

**Comment by Herschel** [comment]
For the bitflags, I think we should move towards storing the flags directly in the structs. For example, `FontInfo` would have `flags: FontFlags`, which would simplify reading/writing and make the structs take less memory. We could provide accessor methods such as `FontInfo::is_bold()` for user code.

File: core/src/display_object/movie_clip.rs (hunk 1 above)
Comment (relrelb) [reply core/src/display_object/movie_clip.rs]: Ack, for now I excluded the relevant commit. I guess we'd want to make this change widely in a separate PR.

**Comment by relrelb** [comment]
> For the bitflags, I think we should move towards storing the flags directly in the structs. For example, `FontInfo` would have `flags: FontFlags`, which would simplify reading/writing and make the structs take less memory. We could provide accessor methods such as `FontInfo::is_bold()` for ease of use in user code.

Nice improvement! I've applied this for `FontFlag` and `FontInfoFlag`. The other `PlaceFlag` and `LineStyleFlag` are more complicated so I remained them unchanged for now.

File: swf/src/write.rs [hunk 3]
```diff
@@ -2244,6 +2173,44 @@ impl<W: Write> Writer<W> {
         Ok(())
     }
 
+    fn write_define_font_info(&mut self, font_info: &FontInfo) -> Result<()> {
+        let use_wide_codes = self.version >= 6 || font_info.version >= 2;
```
Comment (relrelb) [inline swf/src/write.rs:2177]: This is now specified in `self.flags`. We might want to just use `self.flags` instead of this version check.

**Review by Herschel — APPROVED** [review]
Thank you!

**Pull request merged**